docs/nddata/mixins/ndarithmetic.rst

```diff
@@ -1,49 +1,406 @@
+      1
+
+  If one is using `~astropy.wcs.WCS` objects a very handy function to use might
+  be::
+
+      >>> def wcs_compare(wcs1, wcs2, *args, **kwargs):
+      ...     return wcs1.wcs.compare(wcs2.wcs, *args, **kwargs)
+
+  see :meth:`astropy.wcs.Wcsprm.compare` for the arguments this comparison
+  allows.
+
+uncertainty
+^^^^^^^^^^^
+
+the ``propagate_uncertainties`` argument can be used to turn the propagation
+of uncertainties on or off.
+
+- If ``None`` the result will have no uncertainty::
+
+      >>> from astropy.nddata import StdDevUncertainty
+      >>> ndd1 = NDDataRef(1, uncertainty=StdDevUncertainty(0))
+      >>> ndd2 = NDDataRef(1, uncertainty=StdDevUncertainty(1))
+      >>> ndd1.add(ndd2, propagate_uncertainties=None).uncertainty is None
+      True
+
+- If ``False`` this is equivalent to ``"first_found"`` for ``meta``, ... and
+  the result will have the first found uncertainty.
+
+  .. note::
+      Setting ``propagate_uncertainties=False`` is not generally not
+      recommended.
+
+  but it's possible nevertheless::
+
+      >>> ndd1 = NDDataRef(1, uncertainty=StdDevUncertainty([0]))
+      >>> ndd2 = NDDataRef(1, uncertainty=StdDevUncertainty([1]))
+      >>> ndd1.add(ndd2, propagate_uncertainties=False).uncertainty
+      StdDevUncertainty([0])
+
+- If ``True`` both uncertainties must be ``NDUncertainty`` subclasses that
+  implement propagation. This is possible for
+  `~astropy.nddata.StdDevUncertainty`::
+
+      >>> ndd1 = NDDataRef(1, uncertainty=StdDevUncertainty([10]))
+      >>> ndd2 = NDDataRef(1, uncertainty=StdDevUncertainty([10]))
+      >>> ndd1.add(ndd2, propagate_uncertainties=True).uncertainty
+      StdDevUncertainty([ 14.14213562])
+
+uncertainty with correlation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+in case ``propagate_uncertainties`` is ``True`` you can give also an argument
+for ``uncertainty_correlation``. `~astropy.nddata.StdDevUncertainty` cannot
+keep track of it's correlations by itself but it can evaluate the correct
+resulting uncertainty if the correct ``correlation`` is given.
+
+For example without correlation subtracting a `~astropy.nddata.NDDataRef`
+instance from itself results in a non-zero uncertainty::
+
+    >>> ndd1 = NDDataRef(1, uncertainty=StdDevUncertainty([10]))
+    >>> ndd1.subtract(ndd1, propagate_uncertainties=True).uncertainty
+    StdDevUncertainty([ 14.14213562])
+
+but given a correlation of ``1`` because they clearly correlate gives the
+correct uncertainty of ``0``::
+
+    >>> ndd1 = NDDataRef(1, uncertainty=StdDevUncertainty([10]))
+    >>> ndd1.subtract(ndd1, propagate_uncertainties=True,
+    ...               uncertainty_correlation=1).uncertainty
+    StdDevUncertainty([ 0.])
+
+which would be consistent with the equivalent operation ``ndd1 * 0``::
+
+    >>> ndd1.multiply(0, propagate_uncertainties=True).uncertainty
+    StdDevUncertainty([0])
+
+The default (``0``) represents uncorrelated while ``1`` means correlated and
```

Comment:
I think it would be clearer to move this up, right after the sentence that ends "...correct ``correlation`` is given."